app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/SaveMappingMissionActivityTest.kt

```diff
@@ -7,68 +7,110 @@ package ch.epfl.sdp.drone3d.ui.mission
 import androidx.test.espresso.intent.Intents
-import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.espresso.intent.matcher.ComponentNameMatchers
+import androidx.test.espresso.intent.matcher.IntentMatchers
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
 import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.platform.app.InstrumentationRegistry
 import ch.epfl.sdp.drone3d.R
 import ch.epfl.sdp.drone3d.model.auth.UserSession
 import ch.epfl.sdp.drone3d.model.mission.MappingMission
+import ch.epfl.sdp.drone3d.model.weather.WeatherReport
 import ch.epfl.sdp.drone3d.service.api.auth.AuthenticationService
+import ch.epfl.sdp.drone3d.service.api.location.LocationService
 import ch.epfl.sdp.drone3d.service.api.mission.MappingMissionService
 import ch.epfl.sdp.drone3d.service.api.storage.dao.MappingMissionDao
+import ch.epfl.sdp.drone3d.service.api.weather.WeatherService
 import ch.epfl.sdp.drone3d.service.module.AuthenticationModule
+import ch.epfl.sdp.drone3d.service.module.LocationModule
 import ch.epfl.sdp.drone3d.service.module.MappingMissionDaoModule
+import ch.epfl.sdp.drone3d.service.module.WeatherModule
 import com.google.firebase.auth.FirebaseUser
 import com.mapbox.mapboxsdk.geometry.LatLng
 import dagger.hilt.android.testing.BindValue
 import dagger.hilt.android.testing.HiltAndroidRule
 import dagger.hilt.android.testing.HiltAndroidTest
 import dagger.hilt.android.testing.UninstallModules
-import org.hamcrest.Matchers.equalTo
-import org.hamcrest.Matchers.not
 import org.junit.*
 import org.junit.rules.RuleChain
 import org.mockito.Mockito
 import org.mockito.Mockito.*
+import java.lang.Thread.sleep
+import java.util.*
 
 @HiltAndroidTest
-@UninstallModules(AuthenticationModule::class, MappingMissionDaoModule::class)
+@UninstallModules(
+    AuthenticationModule::class,
+    MappingMissionDaoModule::class,
+    LocationModule::class,
+    WeatherModule::class,
+)
 class SaveMappingMissionActivityTest {
 
     companion object {
         private const val USER_UID = "user_id"
+        private val GOOD_WEATHER_REPORT = WeatherReport(
+            "Clear", "description",
+            20.0, 20, 5.0, 500, Date(12903)
+        )
+        private val AREA = arrayListOf(
+            LatLng(46.52109254025395, 6.643008669745938),
+            LatLng(46.52109254025395, 6.64300866974594),
+            LatLng(46.52109254025397, 6.643008669745938),
+        )
+        private val UNNAMED_MISSION = MappingMission(
+            "Unnamed mission",
+            50.0,
+            MappingMissionService.Strategy.SINGLE_PASS,
+            AREA
+        )
+        private val NAMED_MISSION =
+            MappingMission("My mission", 50.0, MappingMissionService.Strategy.SINGLE_PASS, AREA)
+
     }
 
+    @BindValue
+    val locationService: LocationService = Mockito.mock(LocationService::class.java)
```

Comment:
Because you imported Mockito.*, you don't need to specify that mock comes from Mockito
```suggestion
    val locationService: LocationService = mock(LocationService::class.java)
```